modules/javafx.controls/src/main/java/javafx/scene/control/Skinnable.java

```diff
@@ -36,23 +36,30 @@
  */
 public interface Skinnable {
     /**
-     * Skin is responsible for rendering this {@code Control}. From the
-     * perspective of the {@code Control}, the {@code Skin} is a black box.
-     * It listens and responds to changes in state in a {@code Control}.
+     * Skin is responsible for rendering this {@code Skinnable}. From the
+     * perspective of the {@code Skinnable}, the {@code Skin} is a black box.
+     * It listens and responds to changes in state in a {@code Skinnable}.
      * <p>
-     * There is a one-to-one relationship between a {@code Control} and its
+     * There is typically a one-to-one relationship between a {@code Skinnable} and its
      * {@code Skin}. Every {@code Skin} maintains a back reference to the
-     * {@code Control}.
+     * {@code Skinnable}.
      * <p>
      * A skin may be null.
      *
-     * @return the skin property for this control
+     * @return the skin property for this Skinnable
      */
     public ObjectProperty<Skin<?>> skinProperty();
 
     /**
-     * Sets the skin that will render this {@link Control}
-     * @param value the skin value for this control
+     * Sets the skin that will render this {@link Skinnable}.
+     * <p>
+     * To ensure a one-to-one relationship between a {@code Skinnable} and its
+     * {@code Skin}, this method may check the return value of
+     * {@link Skin#getSkinnable()} against this Skinnable,
+     * and may throw an {@code IllegalArgumentException} if it is not the same.
```

Comment:
Since most implementations of `Skinnable` will do this check, we probably want to strengthen this statement. I like the language you added to the `@throws` tag, which says that it will throw an exception if the Skin does not "correspond to" this Skinnable. Is there a way to work this in, by defining what we mean to "correspond to"?

Reply:
@throws IllegalArgumentException if {@link Skin#getSkinnable()} returns a different {@code Skinnable}

is this better?

Reply:
also updated Control skinProperty javadoc, please review.